arangod/Agency/Constituent.cpp

```diff
@@ -261,9 +249,9 @@ void Constituent::lead(term_t term) {
     _recentElections.push_back(readSystemClock());
   }
 
-  // we need to start work as leader
-  _agent->lead();
-  
+  // we need to rebuild spear_head and read_db, but this is done in the
+  // main Agent thread:
+  _agent->beginPrepareLeadership();
```

Comment:
Should this signal the _appendCV?

Reply:
_agent->wakupMainLoop()

(and that inline function should move broadcast() inside LOCKER block.)